src/internal/m365/exchange/events_restore.go

```diff
@@ -210,15 +212,108 @@ func updateExceptionOccurrences(
 
 		evt = toEventSimplified(evt)
 
-		// TODO(meain): Update attachments (might have to diff the
-		// attachments using ids and delete or add). We will have
-		// to get the id of the existing attachments, diff them
-		// with what we need a then create/delete items kinda like
-		// permissions
 		_, err = ac.PatchItem(ctx, userID, ptr.Val(evts[0].GetId()), evt)
 		if err != nil {
 			return clues.Wrap(err, "updating event instance")
 		}
+
+		evt, err = api.EventFromMap(instance)
+		if err != nil {
+			return clues.Wrap(err, "parsing exception event (second)")
+		}
+
+		err = updateAttachments(ctx, ac, userID, containerID, ptr.Val(evts[0].GetId()), evt)
+		if err != nil {
+			return clues.Wrap(err, "updating attachments")
+		}
+	}
+
+	return nil
+}
+
+func updateAttachments(
+	ctx context.Context,
+	client api.Events,
+	userID, containerID, eventID string,
+	event models.Eventable,
+) error {
+	// Attachment can only be deleted or added in modified events and
+	// when an event is modified, the attachment ids will be
+	// changed. This change is both present in the data that we backed
+	// up and so we cannot rely on the attachment id to decide if they
+	// are the same. Instead we make use of the contentBytes and name
+	// to ensure they are the same.
+	attachments, err := client.GetAttachments(ctx, false, userID, eventID)
+	if err != nil {
+		return clues.Wrap(err, "getting attachments")
+	}
+
+	// Delete unnecessary attachments
+	for _, att := range attachments {
```

Comment:
Should we actually delete attachments outside of an on-collision=replace policy?  If we're restoring to a copy, we don't want to delete data from the original.  Or is that isolation already handled at a higher level?

Reply:
copy as in `restore-to-copy` as a restore collision handling option, versus `skip` and `replace`.  It makes sense to delete the original data on a `replace` (on that note, do we need to diff events for instances that have changed in the current version but are not different in the backup?).  But if we're copying the data to a new event entry, should we be deleting the originals?

Reply:
Ahh, I think there has been a misunderstanding here. This is restoring to event exceptions and we are only deleting the ones that we specifically know are deleted from that specific event exception.